src/java/fr/paris/lutece/plugins/workflow/modules/userassignment/service/task/TaskUserAssignmentNotification.java

```diff
@@ -0,0 +1,154 @@
+import fr.paris.lutece.plugins.workflowcore.service.task.TaskService;
+import fr.paris.lutece.portal.business.user.AdminUser;
+import fr.paris.lutece.portal.service.admin.AdminUserService;
+import fr.paris.lutece.portal.service.i18n.I18nService;
+import fr.paris.lutece.portal.service.mail.MailService;
+import fr.paris.lutece.portal.service.template.AppTemplateService;
+import fr.paris.lutece.util.html.HtmlTemplate;
+
+/**
+ * TaskUserAssignmentNotification
+ *
+ */
+public class TaskUserAssignmentNotification extends SimpleTask
+{
+    // Message
+    private static final String MESSAGE_TASK_TITLE 	= "module.workflow.userassignment.task_user_assignment_notification.title";
+
+    //MARKS
+    private static final String MARK_RESOURCE_ID 	= "resourceId";
+    private static final String MARK_RESOURCE_TYPE 	= "resourceType";
+    
+	private static final String BEAN_CONFIG 		= "workflow-userassignment.taskUserAssignmentNotificationConfigService";
+	
+    @Inject
+    @Named( BEAN_CONFIG )
+    private ITaskConfigService _taskConfigService;
+    
+    @Inject
+    @Named( ResourceHistoryService.BEAN_SERVICE )
+    private IResourceHistoryService _resourceHistoryService;
+    
+    @Inject
+    @Named ( TaskService.BEAN_SERVICE )
+    private ITaskService _taskService;
+    
+    @Inject
+    private IAssignUserResourceTaskService _userResourceService;
+    
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getTitle( Locale locale )
+    {
+        return I18nService.getLocalizedString( MESSAGE_TASK_TITLE, locale );
+    }
+
+
+	@Override
+	public void processTask( int nIdResourceHistory, HttpServletRequest request, Locale locale )
+	{
+		ResourceHistory resourceHistory = _resourceHistoryService.findByPrimaryKey( nIdResourceHistory );
+		
+		TaskUserAssignmentNotificationConfig config = _taskConfigService.findByPrimaryKey( getId( ) );
+		
+		if ( resourceHistory != null && config != null )
+		{			
+			List<ITask> listTask = _taskService.getListTaskByIdAction( resourceHistory.getAction( ).getId( ), locale );
+			
+			if ( listTask != null )
+			{
+				for ( ITask task : listTask )
+				{
+					if ( task instanceof AssignUserResourceTask )
+					{
+						notifyUser( request, locale, resourceHistory, config );
+					}
+				}
+			}			
+		}		
+	}
+
+	
+	/**
+	 * Notify user
+	 * @param locale
+	 * @param resourceHistory
+	 * @param config
+	 * @param userConnected
+	 * @param adminUser
+	 */
+	private void notifyUser( HttpServletRequest request, Locale locale, ResourceHistory resourceHistory, TaskUserAssignmentNotificationConfig config )
+	{
+		AdminUser userConnected = AdminUserService.getAdminUser( request );
+
+		for( AdminUser adminUser :  _userResourceService.listActiveUserByResource( resourceHistory.getIdResource( ), resourceHistory.getResourceType( ) ) )
+		{	
+			Map<String, Object> model = new HashMap< >( );
+			
+			model.put( MARK_RESOURCE_ID, resourceHistory.getIdResource( ) );
```

Comment:
What's the use for ?

Reply:
These marks are to be used in the content of the notification message. They will allow us to create URLs for example.